### DISCUSSION TITLE
How to use `current_flow_betweenness_centrality` correctly

### DISCUSSION THREAD

**Discussion by bianchi-dy** [networkx/networkx#5126] [Q&A] [answered]
I am attempting to compute `flow betweenness` using `nx.current_flow_betweenness_centrality(<network>)`. However, I get the following error message: 

```NetworkXError: Graph not connected.```

For reference, I am creating a network of flights between airports, and using a `Graph` to do this (undirected). It currently looks like the image below.

![image](https://user-images.githubusercontent.com/44487900/136765885-b4ac35e9-e825-44aa-a91a-09551cffc842.png)

I posted this in the Q&A section since I'm not sure if I'm using the algorithm correctly; I want to quantify the intensity of 'transfer passenger flows between two airports', using flights as a proxy 'current'. It's based on this paper's approach, pg 4: [The evolution of aviation network: Global airport connectivity index 2006–2016](https://www.sciencedirect.com/science/article/pii/S1366554519301243/pdfft?crasolve=1&iv=a71c138b8b8d63fc71077ab6447840f8&token=6365656331323466303437383261356130626164356433363435323335363666633536636335623762316232326664376165366564316363623536353838656665373764366161313833633938663335373235326230373739323a616464323639613036613336353635343938363866613233&text=0da7de93287e4d9297a10244f15f746521fac8f8199d80c9e263724ca75741972d6aeab79a395029a7dc620c9b09285c2c79a26c53e19df958355a826f8090969cd958dcc840d2e8e84400c7d0f81d5f1befd5c252ef033117f6af55e14a3dd365de79fcd81eecc35691eb9eabc817f440e9928311a8942432b04c3c50530769cbae6b7151540b20e3fa72f2a1ed4a093584c5e63b78dbf0d8d213c037ff68710073370857591f1521a1a3c0d51e1cd551bed01f9b991bfff28af320c02c585e4c8a2dc634e72d25cd5fc4610ee1157a3dd478a22234e3628baebaa83d1594253617902ea35a8cddcc66047de9d29b99c61656191173e47011767a13bbff3ae5d971c9fa8baca40cdf492f79c3c6c639b5ab595cbbed95e0f38cd625609b1b86f6194a32dd88beac20bf4144b7d1077a&original=3f636173615f746f6b656e3d5a335a426161663257593441414141412533415732377374464d372d64634b56676e7a764f4a413569734d5a6b5a366f6a73326a67543933517072736e42324f4e77584d795357415f7941767a2d6c3564644d5538705537455563307235626167266d64353d3731663333643938663536326133363331343336663935376363616439663237267069643d312d73322e302d53313336363535343531393330313234332d6d61696e2e706466)

Is there some transformation or node isolation I need to perform on my network first before applying `flow_betweenness _centrality` in order for the algorithm to work, or am I using the wrong tool for the problem? Is there an alternative I can use within NetworkX? Your help would be much appreciated!

**Comment by MridulS** [comment, marked as answer]
The algorithm `current_flow_betweenness_centrality` only works for connected Graphs, so one way to proceed with this is to clean the dataset and remove all the isolated nodes or just do the analysis on the largest connected component in the `Graph`.
``` python
>>> # assuming the G is the Graph object
>>> cleaned_up_G = G.subgraph(max(nx.connected_components(G), key=len)) 
```
In my experience with aviation networks the isolated airports are usually some "special" use case ones, so it's *should* be okay to remove them from the `Graph`.

**Reply by dschult** [reply]
A performance suggestion about subgraphs. If you plan to do a lot of computation with the resulting subgraph, it is often faster/cheaper to make a copy of the subgraph. That way it doesn't need to check whether the nodes are in the subgraph every time it looks up an edge.  Just add `.copy()` to the end of the suggested line by @MridulS 

```python
>>> cleaned_up_G = G.subgraph(max(nx.connected_components(G), key=len)).copy()
```
